app/controllers/posts_controller.rb

```diff
@@ -725,9 +724,13 @@ def markdown(post)
   # We can't break the API for making posts. The new, queue supporting API
   # doesn't return the post as the root JSON object, but as a nested object.
   # If a param is present it uses that result structure.
-  def backwards_compatible_json(json_obj, success)
+  def backwards_compatible_json(json_obj)
     json_obj.symbolize_keys!
-    if params[:nested_post].blank? && json_obj[:errors].blank? && json_obj[:action] != :enqueued
+
+    success = json_obj[:success]
+
+    if params[:nested_post].blank? && json_obj[:errors].blank? &&
+         json_obj[:action].to_s != "enqueued"
```

Comment:
Here's the actual bug fix, depending on whether we're creating a post "normally" or using the API, the `json_obj` will either be a Ruby object or a json object that has been serialized (for the memoization) and deserialized.

When it's a Ruby object, the `:action` is a symbol and when it's a JSON object, it's a string.

Added the `.to_s` to ensure we handle both cases.